crates/voicevox_core_c_api/tests/e2e/testcases/user_dict_manipulate.rs

```diff
@@ -0,0 +1,222 @@
+            voicevox_user_dict_add_word,
+            voicevox_user_dict_update_word,
+            voicevox_user_dict_remove_word,
+            voicevox_user_dict_get_json,
+            voicevox_user_dict_import,
+            voicevox_user_dict_load,
+            voicevox_user_dict_save,
+            voicevox_user_dict_delete,
+            voicevox_user_dict_uuid_free,
+            voicevox_json_free,
+            ..
+        } = Symbols::new(lib)?;
+
+        let get_json = |dict: &*mut VoicevoxUserDict| -> String {
+            let mut json = MaybeUninit::uninit();
+            assert_ok(voicevox_user_dict_get_json(
+                (*dict) as *const _,
+                json.as_mut_ptr(),
+            ));
+
+            let ret = CStr::from_ptr(json.assume_init())
+                .to_str()
+                .unwrap()
+                .to_string();
+
+            voicevox_json_free(json.assume_init());
+
+            serde_json::from_str::<serde_json::Value>(&ret).expect("invalid json");
+
+            ret
+        };
+
+        let add_word = |dict: &*mut VoicevoxUserDict, word: &VoicevoxUserDictWord| {
+            let mut word_uuid = MaybeUninit::uninit();
+
+            assert_ok(voicevox_user_dict_add_word(
+                (*dict) as *const _,
+                word as *const _,
+                word_uuid.as_mut_ptr(),
+            ));
+
+            let ret = CStr::from_ptr(word_uuid.assume_init())
+                .to_str()
+                .unwrap()
+                .to_string();
+
+            voicevox_user_dict_uuid_free(word_uuid.assume_init());
+
+            ret
+        };
+
+        // テスト用の辞書ファイルを作成
+        let dict = {
+            let mut dict = MaybeUninit::uninit();
+            assert_ok(voicevox_user_dict_new(dict.as_mut_ptr()));
+            dict.assume_init()
+        };
+
+        // 単語の追加のテスト
+        let word = {
+            let mut word = voicevox_default_user_dict_word();
+            word.surface = CString::new("hoge").unwrap().into_raw();
+            word.pronunciation = CString::new("ホゲ").unwrap().into_raw();
+            word.word_type = VoicevoxUserDictWordType::VOICEVOX_USER_DICT_WORD_TYPE_PROPER_NOUN;
+
+            word
+        };
+
+        let word_uuid = add_word(&dict, &word);
+
+        let json = get_json(&dict);
+
+        assert!(json.contains("ｈｏｇｅ"));
+        assert!(json.contains("ホゲ"));
+        assert!(json.contains(word_uuid.as_str()));
+
+        // 単語の変更のテスト
+        let word = {
+            let mut word = voicevox_default_user_dict_word();
+            word.surface = CString::new("fuga").unwrap().into_raw();
+            word.pronunciation = CString::new("フガ").unwrap().into_raw();
+            word.word_type = VoicevoxUserDictWordType::VOICEVOX_USER_DICT_WORD_TYPE_COMMON_NOUN;
+
+            word
+        };
+
+        assert_ok(voicevox_user_dict_update_word(
+            dict,
+            word_uuid.as_bytes().as_ptr() as *const i8,
+            &word,
+        ));
```

Comment:
ここでは...?
(`word_uuid`はnul-terminatedじゃないRustの`String`)

```suggestion
        assert_ok({
            let word_uuid = CString::new(&*word_uuid).unwrap();
            voicevox_user_dict_update_word(dict, word_uuid.as_ptr(), &word)
        });
```

Reply:
↑のsuggestionよりは、`word_uuid`を`CString`で持っといた方がいいかも。